From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 23 Aug 2016 15:48:56 +0200
Subject: [PATCH] fpb: Support multiple versions of packages

This is a temporary change until the fpm installed by pip gets
the change from [1] included.

[1] https://review.openstack.org/#/c/311031/

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 build/docker/Dockerfile | 9 ++++++++-
 build/docker/Makefile   | 4 ++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/build/docker/Dockerfile b/build/docker/Dockerfile
index 624f233..ab0d0d3 100644
--- a/build/docker/Dockerfile
+++ b/build/docker/Dockerfile
@@ -23,7 +23,14 @@ RUN apt-get install -y software-properties-common python-software-properties \
     build-essential ruby-dev rubygems-integration python-pip git rpm createrepo dpkg-dev
 
 RUN gem install fpm
-RUN pip install fuel-plugin-builder
+
+# Temporary: fpb needs to be built from sources
+# RUN pip install fuel-plugin-builder
+RUN git clone -b INSERT_FPB_BRANCH INSERT_FPB_REPO && cd fuel-plugins && \
+    (test -z INSERT_FPB_CHANGE || \
+        (git fetch origin INSERT_FPB_CHANGE && git checkout FETCH_HEAD)) && \
+    python setup.py sdist && pip install ./dist/fuel-plugin-builder-*.tar.gz && \
+    cd .. && rm -rf fuel-plugins
 
 RUN echo "ALL ALL=NOPASSWD: ALL" > /etc/sudoers.d/open-sudo
 RUN echo "Defaults env_keep += \"ftp_proxy http_proxy https_proxy no_proxy RSYNC_PROXY RSYNC_CONNECT_PROG npm_config_registry\"" > /etc/sudoers.d/keep-proxies
diff --git a/build/docker/Makefile b/build/docker/Makefile
index d4423b0..226bbd0 100644
--- a/build/docker/Makefile
+++ b/build/docker/Makefile
@@ -25,4 +25,8 @@ all:	.docker
 	cp Dockerfile ubuntu-builder/Dockerfile
+	# Only add FPB ENVs when set - needed to fetch, patch and install FPB
+	test -n "${FPB_REPO}" && sed -i "s;INSERT_FPB_REPO;${FPB_REPO};" ubuntu-builder/Dockerfile || exit 0
+	test -n "${FPB_BRANCH}" && sed -i "s;INSERT_FPB_BRANCH;${FPB_BRANCH};" ubuntu-builder/Dockerfile || exit 0
+	test -n "${FPB_CHANGE}" && sed -i "s;INSERT_FPB_CHANGE;${FPB_CHANGE};" ubuntu-builder/Dockerfile || exit 0
 	# Only add proxy ENVs where set in host - needed to pull the base Ubuntu image
 	test -n "${http_proxy}" && sed -i "s;INSERT_HTTP_PROXY;${http_proxy};" ubuntu-builder/Dockerfile || exit 0
 	test -n "${https_proxy}" && sed -i "s;INSERT_HTTPS_PROXY;${https_proxy};" ubuntu-builder/Dockerfile || exit 0
